From 6f1c0ccabfe57541390a92e4d2f0470c772a8e19 Mon Sep 17 00:00:00 2001
From: Youngjin Kim <youngjin78.kim@lge.com>
Date: Wed, 14 Aug 2019 08:25:49 +0900
Subject: [PATCH] v3d: Fix screen artifacts when using EGL image by
 EGL_WAYLAND_BUFFER_WL

When retrieve DRI image from file description by dri2_from_fds, it always pass DRM_FORMAT_MOD_INVALID for modifier.
vc4 can decide from vc4_ioctl whether it was tiled or not, but v3d doesn't have any way to decide it now.
Previously, it was treated as tiled when v3d_resource_from_handle with DRM_FORMAT_MOD_INVALID.
But it was changed in order to fix DRI3 rendering with RO on X11.
Now it will check screen has render only property(ro) if use X11 platform, otherwise resource treats as tiled.
---
 src/gallium/drivers/v3d/v3d_resource.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/gallium/drivers/v3d/v3d_resource.c b/src/gallium/drivers/v3d/v3d_resource.c
index 10abee4..b809290 100644
--- a/src/gallium/drivers/v3d/v3d_resource.c
+++ b/src/gallium/drivers/v3d/v3d_resource.c
@@ -830,7 +830,11 @@ v3d_resource_from_handle(struct pipe_screen *pscreen,
                 rsc->tiled = true;
                 break;
         case DRM_FORMAT_MOD_INVALID:
+ #ifdef HAVE_X11_PLATFORM
                 rsc->tiled = screen->ro == NULL;
+ #else
+                rsc->tiled = true;
+ #endif
                 break;
         default:
                 fprintf(stderr,
-- 
2.7.4

